sdks/go/pkg/beam/core/runtime/exec/translate.go

```diff
@@ -588,6 +592,20 @@ func (b *builder) makeLink(from string, id linkID) (Node, error) {
 						}
 					}
 
+					if len(userTimers) > 0 {
+						log.Debugf(context.TODO(), "userTimers %+v", userTimers)
+						timerIDToCoder := make(map[string]*coder.Coder)
+						sID := StreamID{Port: Port{URL: b.desc.GetTimerApiServiceDescriptor().GetUrl()}, PtransformID: id.to}
+						ec, wc, err := b.makeCoderForPCollection(input[0])
+						if err != nil {
+							return nil, err
+						}
+						for key := range userTimers {
+							timerIDToCoder[key] = coder.NewT(ec, wc)
+						}
```

Comment:
2. Yeah, here we don't even use the coderID stored in the spec, but we ultimately pass through the whole PCollection coder instead of extracting out the Key component here.

Reply:
Good catch, I agree. We should just send the key coder from here which then simplifies the exec/coder.go for timer encoding/decoding